components/ruby/lib/chef-licensing/license_key_fetcher/file.rb

```diff
@@ -271,5 +291,105 @@ def handle_error(e, message = nil)
         e
       end
     end
+
+    class LicenseFile
+      def self.verify_structure(data, expected_structure)
+        return false unless data.is_a?(Hash)
+
+        expected_structure.each do |key, value|
+          return false unless data.key?(key)
+
+          if value.is_a?(Hash)
+            return false unless verify_structure(data[key], value)
+          elsif value.is_a?(Array)
+            return false unless data[key].is_a?(Array)
+
+            data[key].each do |item|
+              return false unless verify_structure(item, value[0])
+            end
+          elsif value.is_a?(Class)
+            return false unless data[key].is_a?(value)
+          else
+            return false unless data[key] == value
+          end
+        end
+
+        true
+      end
+
+      def self.load_primary_structure
+        raise NotImplementedError
+      end
+    end
+
+    class V4LicenseFile < LicenseFile
```

Comment:
and these can be in its separate files

Reply:
also my suggestion would be to namespace the versions rather than prefixing classname itself